packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -65,7 +69,10 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
     override async showSaveDialog(props: SaveFileDialogProps, folder?: FileStat): Promise<URI | undefined> {
         const rootNode = await this.getRootNode(folder);
         if (rootNode) {
-            const { filePath } = await electronRemote.dialog.showSaveDialog(this.toSaveDialogOptions(rootNode.uri, props));
+            const electronWindow = electronRemote.getCurrentWindow();
+            const isModal = props.modal !== false;
+            const { filePath } = isModal ? await electronRemote.dialog.showSaveDialog(electronWindow, this.toSaveDialogOptions(rootNode.uri, props)) :
+                await electronRemote.dialog.showSaveDialog(this.toSaveDialogOptions(rootNode.uri, props));
```

Comment:
same is true as above comment, with additional formatting:

```suggestion
            const { filePath } = props.modal !== false ? 
                await electronRemote.dialog.showSaveDialog(electronRemote.getCurrentWindow(), this.toSaveDialogOptions(rootNode.uri, props)) :
                await electronRemote.dialog.showSaveDialog(this.toSaveDialogOptions(rootNode.uri, props));
```